core/src/main/java/org/apache/james/core/Username.java

```diff
@@ -125,6 +125,17 @@ public String asId() {
         return asString().toLowerCase(Locale.US);
     }
 
+    public boolean equalsAsId(String otherId) {
```

Comment:
is there any reason you don't override equals() but creating this method?

Reply:
I can answer you: username is case sensitive, it is its usage within UsersReposiotory which is case insentive, hence we should not alter equals with such concerns (source of potential wrong tests and thus bugs)

Reply:
I still don't feel comfortable with more than one equals... method in a POJO, should the `Id` be strong typed?

Reply:
POJO is the right place to put some domain logic.